### ISSUE TITLE
Arbitrary Invoice.reference max length causing ValueError

### ISSUE THREAD

**Issue by sromero84** [XeroAPI/xero-python#34]
The `Invoice.reference` setter adds a constraint to limit the length to `255`, which is not stated in Xero API documentation,, not restricted anyway from the API endpoints or the UI.

* Client library limit: https://github.com/XeroAPI/xero-python/blob/master/xero_python/accounting/models/invoice.py#L471-L486
* Xero API documentation: https://developer.xero.com/documentation/api/invoices
* Invoice created in the UI with reference value larger than `255`:

![Screen Shot 2020-12-04 at 17 57 59](https://user-images.githubusercontent.com/1017709/101214833-7357f480-365b-11eb-8760-06e91730c9dd.png)

The reference value is exacly `56639794187018499005505537051269050269031747341705170869956784690304435842538098469025249022458498592412373024709304465074886319553161205154991120968889176852720732909084251608420103323043086720381912246979450257446135888016873358122462450888111033111060942298` and `260` characters long.

I've run into this issue while trying to pull data from a customer's account.

I'd like to confirm this is a potential issue for many other users, and provide a pull request removing this max length (or adding a much larger limit) if needed.

The logs traceback is:

```
[2020-12-04 20:04:22,380] {pod_launcher.py:156} INFO - b'  File "/usr/local/lib/python3.7/site-packages/xero_python/accounting/api/accounting_api.py", line 10130, in get_invoices\n'
[2020-12-04 20:04:22,380] {pod_launcher.py:156} INFO - b'    collection_formats=collection_formats,\n'
[2020-12-04 20:04:22,380] {pod_launcher.py:156} INFO - b'  File "/usr/local/lib/python3.7/site-packages/xero_python/api_client/__init__.py", line 365, in call_api\n'
[2020-12-04 20:04:22,380] {pod_launcher.py:156} INFO - b'    _request_timeout,\n'
[2020-12-04 20:04:22,380] {pod_launcher.py:156} INFO - b'  File "/usr/local/lib/python3.7/site-packages/xero_python/api_client/__init__.py", line 214, in __call_api\n'
[2020-12-04 20:04:22,381] {pod_launcher.py:156} INFO - b'{"logger": "urllib3.connectionpool", "timestamp": "2020-12-04 20:04:22", "severity": "debug", "message": "Starting new HTTPS connection (1): o374532.ingest.sentry.io:443"}\n'
[2020-12-04 20:04:22,381] {pod_launcher.py:156} INFO - b'    response_data, response_type, response_model_finder\n'
[2020-12-04 20:04:22,381] {pod_launcher.py:156} INFO - b'  File "/usr/local/lib/python3.7/site-packages/xero_python/api_client/__init__.py", line 246, in deserialize\n'
[2020-12-04 20:04:22,381] {pod_launcher.py:156} INFO - b'    return deserialize(response_type, data, response_model_finder)\n'
[2020-12-04 20:04:22,381] {pod_launcher.py:156} INFO - b'  File "/usr/local/lib/python3.7/site-packages/xero_python/single_dispatch_str.py", line 90, in wrapper\n'
[2020-12-04 20:04:22,381] {pod_launcher.py:156} INFO - b'    return dispatch(key)(*args, **kw)\n'
[2020-12-04 20:04:22,381] {pod_launcher.py:156} INFO - b'  File "/usr/local/lib/python3.7/site-packages/xero_python/api_client/deserializer.py", line 58, in deserialize\n'
[2020-12-04 20:04:22,381] {pod_launcher.py:156} INFO - b'    return deserialize_model(model, data, model_finder)\n'
[2020-12-04 20:04:22,381] {pod_launcher.py:156} INFO - b'  File "/usr/local/lib/python3.7/site-packages/xero_python/api_client/deserializer.py", line 281, in deserialize_model\n'
[2020-12-04 20:04:22,381] {pod_launcher.py:156} INFO - b'    kwargs[attr] = deserialize(attr_type, value, model_finder)\n'
[2020-12-04 20:04:22,381] {pod_launcher.py:156} INFO - b'  File "/usr/local/lib/python3.7/site-packages/xero_python/single_dispatch_str.py", line 90, in wrapper\n'
[2020-12-04 20:04:22,382] {pod_launcher.py:156} INFO - b'    return dispatch(key)(*args, **kw)\n'
[2020-12-04 20:04:22,382] {pod_launcher.py:156} INFO - b'  File "/usr/local/lib/python3.7/site-packages/xero_python/api_client/deserializer.py", line 77, in deserialize_list\n'
[2020-12-04 20:04:22,382] {pod_launcher.py:156} INFO - b'    return [deserialize(sub_data_type, sub_data, model_finder) for sub_data in data]\n'
[2020-12-04 20:04:22,382] {pod_launcher.py:156} INFO - b'  File "/usr/local/lib/python3.7/site-packages/xero_python/api_client/deserializer.py", line 77, in <listcomp>\n'
[2020-12-04 20:04:22,382] {pod_launcher.py:156} INFO - b'    return [deserialize(sub_data_type, sub_data, model_finder) for sub_data in data]\n'
[2020-12-04 20:04:22,382] {pod_launcher.py:156} INFO - b'  File "/usr/local/lib/python3.7/site-packages/xero_python/single_dispatch_str.py", line 90, in wrapper\n'
[2020-12-04 20:04:22,382] {pod_launcher.py:156} INFO - b'    return dispatch(key)(*args, **kw)\n'
[2020-12-04 20:04:22,382] {pod_launcher.py:156} INFO - b'  File "/usr/local/lib/python3.7/site-packages/xero_python/api_client/deserializer.py", line 58, in deserialize\n'
[2020-12-04 20:04:22,382] {pod_launcher.py:156} INFO - b'    return deserialize_model(model, data, model_finder)\n'
[2020-12-04 20:04:22,382] {pod_launcher.py:156} INFO - b'  File "/usr/local/lib/python3.7/site-packages/xero_python/api_client/deserializer.py", line 283, in deserialize_model\n'
[2020-12-04 20:04:22,383] {pod_launcher.py:156} INFO - b'    instance = model(**kwargs)\n'
[2020-12-04 20:04:22,383] {pod_launcher.py:156} INFO - b'  File "/usr/local/lib/python3.7/site-packages/xero_python/accounting/models/invoice.py", line 217, in __init__\n'
[2020-12-04 20:04:22,383] {pod_launcher.py:156} INFO - b'    self.reference = reference\n'
[2020-12-04 20:04:22,383] {pod_launcher.py:156} INFO - b'  File "/usr/local/lib/python3.7/site-packages/xero_python/accounting/models/invoice.py", line 482, in reference\n'
[2020-12-04 20:04:22,383] {pod_launcher.py:156} INFO - b'    "Invalid value for `reference`, "\n'
[2020-12-04 20:04:22,383] {pod_launcher.py:156} INFO - b'ValueError: Invalid value for `reference`, length must be less than or equal to `255`\n'

```

**Comment by SidneyAllen** [comment]
@sromero84 - Thanks for reporting this issue and the offer to create a PR.

Since our SDKs are generated from our OpenAPI spec, we'll need to modify the spec so this change will take effect in the next build of the xero-python sdk.  

I'll keep this issue open until a fix has been released

**Comment by sromero84** [comment]
@SidneyAllen Thanks.

**Comment by SidneyAllen** [comment]
@sromero84 - this issue resolved in version 1.2.0 of xero-python SDK
https://github.com/XeroAPI/xero-python/releases/tag/1.2.0
